examples/threading/threading.c

```diff
@@ -8,12 +8,38 @@
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
+#define MILLI_TO_MICRO 1000
+
 void* threadfunc(void* thread_param)
 {
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *)thread_param;
+    
+    //sleep after converting milliseconds to microseconds
+    usleep(thread_func_args->wait_to_obtain_ms * MILLI_TO_MICRO);
+    
+    //lock mutex and check return value
+    if (0 != (pthread_mutex_lock(thread_func_args->mutex))) 
+    {
+        perror("ERROR: pthread_mutex_lock failed");
+        exit(-1);
```

Comment:
@cchoi22915 what does it mean to exit() here?  Is terminating the process the right thing to do when there's an error in the thread?